cpp/src/parquet/statistics.h

```diff
@@ -372,11 +445,14 @@ template <typename DType>
 std::shared_ptr<TypedStatistics<DType>> MakeStatistics(
     const ColumnDescriptor* descr, const std::string& encoded_min,
     const std::string& encoded_max, int64_t num_values, int64_t null_count,
-    int64_t distinct_count, bool has_min_max, bool has_null_count,
-    bool has_distinct_count, ::arrow::MemoryPool* pool = ::arrow::default_memory_pool()) {
-  return std::static_pointer_cast<TypedStatistics<DType>>(Statistics::Make(
-      descr, encoded_min, encoded_max, num_values, null_count, distinct_count,
-      has_min_max, has_null_count, has_distinct_count, pool));
+    int64_t distinct_count, const EncodedGeometryStatistics& geometry_statistics,
```

Comment:
Please move `has_geometry_statistics` and `geometry_statistics` to the end and add default values so we don't break public API.

Reply:
I've added a new overload version of this function while preserving the existing one to keep public API compatibility.

Reply:
Why not directly add `const EncodedGeometryStatistics* geometry_statistics = NULLPTR` to the end of the existing function signature?

Reply:
I thought that keeping `::arrow::MemoryPool* pool` at the end of the parameter list is better than interleaving it in between statistics value parameters. I'll switch away from the function overloading approach if you prefer adding geometry_statistics with default values to the end of the parameter list.

Reply:
IMO, we'd better not to add another overload. If there is a compelling reason to do so, we can add a `ARROW_DEPRECATED` macro to the old one.